plugin/storage/s3/s3.go

```diff
@@ -0,0 +1,85 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	awscredentials "github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+// Credentials is the AWS S3 credentials.
+type Credentials struct {
+	AccessKeyID     string `json:"accessKeyId"`
+	SecretAccessKey string `json:"secretAccessKey"`
+}
+
+// Client wraps the AWS S3 client.
+type Client struct {
+	c      *s3.Client
+	bucket string
+}
+
+// NewClient returns a new AWS S3 client.
+func NewClient(ctx context.Context, region, bucket string, credentials Credentials) (*Client, error) {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx,
+		awsconfig.WithRegion(region),
+		awsconfig.WithCredentialsProvider(awscredentials.NewStaticCredentialsProvider(credentials.AccessKeyID, credentials.SecretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS S3 config, error: %w", err)
+	}
+	s3client := s3.NewFromConfig(cfg)
+	client := &Client{
+		c:      s3client,
+		bucket: bucket,
+	}
+	return client, nil
+}
+
+// ListObjects lists objects with prefix in their names.
+func (c *Client) ListObjects(ctx context.Context, prefix string) (*s3.ListObjectsV2Output, error) {
```

Comment:
I may just call

List
Read
Create
Delete

the storage interface doesn't assume client/server arch

Reply:
I prefer to delay the interface design after we implemented the backup and restore logic, including PITR.
The whole idea SGTM.

Reply:
Is this ready for review?

Reply:
Yes.